x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/BinarySpatialFunction.java

```diff
@@ -150,7 +156,13 @@ protected TypeResolution resolveType() {
         protected Expression.TypeResolution isSpatial(Expression e, TypeResolutions.ParamOrdinal paramOrd) {
```

Comment:
Making `isSpatial` depend on the function is confusing to me. There is another `isSpatial` method in DataType.java and this has a fixed meaning - now we make the meaning depending on the `supportsGrid` attribute. Maybe this method deserves a better name.

Reply:
It seems this function has expanded in meaning in a few steps, so the name has become less appropriate. Already in June 2024, when we added the distance function, it was more complex. I could fix it in this PR, or we treat that separately.

Reply:
Renamed to `isCompatibleSpatial`